### PR TITLE
Merged ResourceEvents with ModuleEvents.

### PR DESCRIPTION
**Pull request by devildrake**
Renamed ModuleEventSystem to ModuleEvents.
Re estructured some of the ways in which EventTypes were named.

### CODE REVIEW

File: Project/Source/Event.h [hunk 1]
```diff
@@ -2,25 +2,45 @@
 
 #include "GameObject.h"
 
+class Resource;
+
```
Comment (JaimeBea) [inline Project/Source/Event.h:6]: Falta un 'struct AssetFolder;'

File: Project/Source/Modules/ModuleEvents.h [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	void ProcessEvent(Event& e);
+	std::unordered_map<EventType, std::vector<Module*>> observerMap;
+
+private:
+	Event PopEvent();
+	void ProcessEvents();
+
+public:
+	ModuleEvents();
+	~ModuleEvents();
```
Comment (JaimeBea) [inline Project/Source/Modules/ModuleEvents.h:22]: Constructor y destructor se pueden eliminar

File: Project/Source/Modules/ModuleEvents.h [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+class ModuleEvents : public Module {
+private:
+	concurrency::concurrent_queue<Event> eventQueue;
+	void ProcessEvent(Event& e);
+	std::unordered_map<EventType, std::vector<Module*>> observerMap;
+
+private:
+	Event PopEvent();
+	void ProcessEvents();
```
Comment (JaimeBea) [inline Project/Source/Modules/ModuleEvents.h:18]: 'public' va antes que 'private'

File: Project/Source/Modules/ModuleEvents.h [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+#include "Modules/Module.h"
+
+#include <concurrent_queue.h>
+#include <vector>
+#include <unordered_map>
+
+class ModuleEvents : public Module {
+private:
+	concurrency::concurrent_queue<Event> eventQueue;
+	void ProcessEvent(Event& e);
```
Comment (JaimeBea) [inline Project/Source/Modules/ModuleEvents.h:13]: Mete ProcessEvent con el resto de métodos privados

File: Project/Source/Modules/ModuleResources.h [hunk 5]
```diff
@@ -4,45 +4,33 @@
...
+//enum class ResourceEventType {
+//	ADD_RESOURCE,
+//	REMOVE_RESOURCE,
+//	UPDATE_FOLDERS
+//};
+//
+//struct ResourceEvent {
+//	ResourceEventType type = ResourceEventType::ADD_RESOURCE;
+//	void* object = nullptr;
+//};
```
Comment (JaimeBea) [inline Project/Source/Modules/ModuleResources.h:25]: Comentario innecesario

File: Project/Source/Modules/ModuleResources.h [hunk 6]
```diff
@@ -64,13 +52,14 @@ class ModuleResources : public Module {
...
 	Resource* CreateResourceByType(ResourceType type, const char* assetFilePath, UID id);
+	void SendAddResourceEvent(Resource* resource);
 
 private:
 	std::unordered_map<UID, Resource*> resources;
 	std::unordered_map<UID, unsigned> referenceCounts;
 	AssetFolder* rootFolder = nullptr;
 
-	concurrency::concurrent_queue<ResourceEvent> resourceEventQueue;
+	//concurrency::concurrent_queue<ResourceEvent> resourceEventQueue;
```
Comment (JaimeBea) [inline Project/Source/Modules/ModuleResources.h:62]: Comentario innecesario

File: Project/Source/Modules/ModuleResources.h [hunk 7]
```diff
@@ -80,6 +69,8 @@ template<typename T>
 inline T* ModuleResources::CreateResource(const char* assetFilePath, UID id) {
 	std::string resourceFilePath = GenerateResourcePath(id);
 	T* resource = new T(id, assetFilePath, resourceFilePath.c_str());
-	resourceEventQueue.push({ResourceEventType::ADD_RESOURCE, resource});
+	//resourceEventQueue.push({ResourceEventType::ADD_RESOURCE, resource});
```
Comment (JaimeBea) [inline Project/Source/Modules/ModuleResources.h:72]: Comentario innecesario

**Review by JaimeBea — APPROVED** [review]

File: Project/Source/Modules/ModuleResources.h (hunk 6 above)
Comment (devildrake) [reply Project/Source/Modules/ModuleResources.h]: Fixed.

File: Project/Source/Modules/ModuleResources.h (hunk 7 above)
Comment (devildrake) [reply Project/Source/Modules/ModuleResources.h]: Fixed.

File: Project/Source/Modules/ModuleResources.h (hunk 5 above)
Comment (devildrake) [reply Project/Source/Modules/ModuleResources.h]: Fixed.

File: Project/Source/Modules/ModuleEvents.h (hunk 4 above)
Comment (devildrake) [reply Project/Source/Modules/ModuleEvents.h]: Fixed.

File: Project/Source/Modules/ModuleEvents.h (hunk 3 above)
Comment (devildrake) [reply Project/Source/Modules/ModuleEvents.h]: Fixed.

File: Project/Source/Modules/ModuleEvents.h (hunk 2 above)
Comment (devildrake) [reply Project/Source/Modules/ModuleEvents.h]: Fixed.

File: Project/Source/Event.h (hunk 1 above)
Comment (devildrake) [reply Project/Source/Event.h:7]: Fixed.

**Review by Jordi-Gil — APPROVED** [review]

**Pull request merged**